modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -886,6 +887,116 @@ def test_reset_index(data):
     df_equals(modin_df_cp, pd_df_cp)
 
 
+@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
```

Comment:
this PR adds 4900 new tests only for `reset_index` (previously number of tests for the whole DataFrame API was about ~2200), so the time of running `test_indexing.py` increased a lot, here is a time comparison of executing `test_indexing.py` for a few of our GH-actions job
```
                                            master    #2558
test-ubuntu (engine python, python 3.6)     49s       17m 33s
test-ubuntu (engine dask, python 3.7)       1m 56s    1h 0m 20s
``` 
we should decrease the amount of tests for `reset_index` somehow

Reply:
You don't have to run all these tests simultaneously. When I run them on 56 cores system with `-n 56` I get 20k tests executed in 2 minutes.

Reply:
But it will be just impossible to run all of the `reset_index` tests for external contributors on their home machine. I've ran these tests with 24 cores and it took me about 9 minutes to execute them all (vs few seconds previously). Imagine how frustrating it will be to rerun these tests locally again and again while changing/debugging `reset_index` functionality, while not having 56 cores. I strongly disagree about adding that big amount of tests. I believe that we can cover the same level of functionality by the significantly less amount of tests.